gslib/wildcard_iterator.py

```diff
@@ -643,6 +650,10 @@ def __iter__(self, bucket_listing_fields=None):
     for filepath in filepaths:
       expanded_url = StorageUrlFromString(filepath)
       try:
+        if self.exclude_pattern and self.exclude_pattern.match(filepath):
```

Comment:
The help text mentions that the pattern is relative to the source https://cloud.google.com/storage/docs/gsutil/commands/rsync#options

But I guess in this case ,let's say I have a pattern `data/.*` and the source is `dir`, I think `the expanded_url` will be `dir/data/a.txt` but the pattern will be `data/.*` and it won't match?

Reply:
I saw this too... I refactored the code to match the logic in rsync 1:1 so that skips happen in the wildcard iterator first.